Doc/whatsnew/3.10.rst

```diff
@@ -725,6 +725,10 @@ Removed
   (Contributed by Yurii Karabas, Andrew Svetlov, Yury Selivanov and Kyle Stanley
   in :issue:`42392`.)
 
+* Prohibited passing non-:class:`concurrent.futures.ThreadPoolExecutor`
```

Comment:
Move it to 3.11. Also, I think that would be better to place it in the "Changes in the Python API" section instead of the "Removed" section.

Reply:
Done, thanks